src/core/messages.ts

```diff
@@ -542,7 +543,7 @@ export class DefaultMessages implements Messages {
     const timestamp = new Date(response.timestamp);
     return new DefaultMessage({
       serial: response.serial,
-      clientId: this._clientId,
+      clientId: this._clientId.get(),
```

Comment:
You're allowed to send a message without establishing a connection, right? So this might unnecessarily fail given that the resolver depends the establishment of a realtime connection.

By the time that the REST API call has returned, the core SDK should have fetched a token (since it needs one to make the REST request), so I think that if you change the resolver to not wait for a connection and instead just always re-fetch the client ID from the realtime client then it should work.

(will look for a core SDK spec point that confirms this)

Reply:
I've actually got a separate PR to get this information from the server response directly, so this will go away shortly anyway!

Reply:
So interestingly, this is a problem for JWTs. If you're using a JWT then you have two options:

- Force a `clientId` in `ClientOptions`
- Connect to Ably first

As `TokenRequest` and `TokenDetails` aren't used in that flow